### ISSUE TITLE
Tagged tasks won't get renamed when using task_run_name

### ISSUE THREAD

**Issue by sti0** [PrefectHQ/prefect#8510]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

Tagged tasks are not renamed when using the `task_run_name` option.
![image](https://user-images.githubusercontent.com/18382402/218678605-d5fea073-1734-45af-aee3-b51075fdd7c6.png)

### Reproduction

```python3
from prefect import flow, task
import time


@task(task_run_name="Hey {name}")
def testtask(name):
    time.sleep(10)


@flow
def main():
    names = ["Julia", "Arnold", "Timmy", "Peter", "Susan"]
    n1 = testtask.map(names)
    testtask.with_options(tags=["naming"]).map(names, wait_for=[n1])


main()
```

### Versions

```Text
Version:             2.8.0
API version:         0.8.4
Python version:      3.10.9
Git commit:          50f0ad0a
Built:               Thu, Feb 9, 2023 3:38 PM
OS/Arch:             linux/x86_64
Profile:             postgres
Server type:         hosted
```

**Comment by serinamarie** [comment]
I believe this is a duplicate of an existing issue: https://github.com/PrefectHQ/prefect/issues/8480

**Comment by cicdw** [comment]
This behavior has been fixed in 3.0: regardless of whether a task is cached or not, it's run will be properly renamed.
